src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String COMMA_SEPARATOR = ",";
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        WorkWithFile report = new WorkWithFile();
+        report.writeReportToFile(createReport(readFromFile(fromFileName)), toFileName);
+    }
+
+    private String readFromFile(String fileName) {
+        File fileRead = new File(fileName);
+        StringBuilder result = new StringBuilder();
+        try (BufferedReader bf = new BufferedReader(new FileReader(fileRead))) {
+            String bufferValue = bf.readLine();
+            while (bufferValue != null) {
+                result.append(bufferValue).append(System.lineSeparator());
+                bufferValue = bf.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file" + fileName, e);
+        }
+        return result.toString();
+    }
+
+    private String[] createReport(String dataFromFile) {
+        int supply = 0;
+        int buy = 0;
+        for (String string: dataFromFile.split("\\s+")) {
+            if (string.contains(SUPPLY)) {
+                supply += Integer.parseInt(string.split(COMMA_SEPARATOR)[1]);
+            } else {
+                buy += Integer.parseInt(string.split(COMMA_SEPARATOR)[1]);
+            }
+        }
+        StringBuilder supplyBuilder = new StringBuilder().append(SUPPLY)
```

Comment:
It's better to return just String and don't create so many StringBuilder objects
```
result = supplyResult - buyResult;
StringBuilder builder = new StringBuilder();
        builder.append(SUPPLY).append(COMMA).append(supplyResult).append(System.lineSeparator())
                .append(BUY).append(COMMA).append(buyResult).append(System.lineSeparator())
                .append(RESULT).append(COMMA).append(result);
        return builder.toString();
```